src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -601,7 +618,16 @@ public override Task WriteAsync(Byte[] array, int offset, int count, Cancellatio
             Interlocked.Increment(ref _asyncOperations);
             try
             {
-                await base.WriteAsync(array, offset, count, cancellationToken).ConfigureAwait(false);
+                EnsureNotDisposed();
+
+                await WriteDeflaterOutputAsync(cancellationToken).ConfigureAwait(false);
+
+                // Pass new bytes through deflater
+                _deflater.SetInput(array, offset, count);
+
+                await WriteDeflaterOutputAsync(cancellationToken).ConfigureAwait(false);
+
+                _wroteBytes = true;
```

Comment:
What happens if we successfully wrote via the first WriteDeflatorOutputAsync, but then the second one ends up throwing an exception... looks like we won't set _wroteBytes to true, even though we did?